src/yadg/extractors/__init__.py

```diff
@@ -55,12 +56,54 @@ def extract(
 
 
 def extract_from_path(
-    path: str,
+    source: Path | str,
     extractor: FileType,
+    path: str = None,
 ) -> DataTree:
     """
     Extracts data and metadata from the provided path using the supplied extractor.
 
+    The individual extractor functionality of yadg is called from here. The data is
+    always returned as a :class:`DataTree`. The ``original_metadata`` entries in
+    the returned objects are flattened using json serialisation. The returned objects
+    have a :func:`to_netcdf` as well as a :func:`to_dict` method, which can be used to
+    write the returned object into a file.
+    """
+    if path is not None:
+        logger.warning(
+            "The parameter 'path' is deprecated and has been replaced by 'source'. "
+            "Please use 'source' instead.",
+            DeprecationWarning,
+        )
+        source = path
```

Comment:
This should probably be handled via a decorator, otherwise the function will fail when `source` is not specified. We can then re-use the decorator to process then `fn -> source` transition in each `extract()` function of the individual extractors.